code/datums/mind.dm

```diff
@@ -629,6 +629,31 @@
 	for(var/obj/effect/proc_holder/S in spell_list)
 		RemoveSpell(S)
 
+/datum/mind/proc/rot_mind() //most rotten spells also swap the victims spells for more rot spells, this proc handles that.
+	if(!spell_list.len)
+		return
+	var/spells_to_rot = 0 //how many rot spells to add.
+	for(var/obj/effect/proc_holder/spell/spell in spell_list)
+		spells_to_rot++
+		RemoveSpell(spell)

```

Comment:
Add a can_rot var to spells, which is set to 0 (in order to automatically exclude racial spells). You will, sadly, have to set it to 1 for each spell that can rot individually. On this loop, you will create a list of spells that can rot and after that loop, RemoveSpell(pick(rotting_spells)) after a for loop depending on spells_to_rot. That way you can solve Ath's problem and also solve rotting spells removing rotting spells (which would be redundant).

Reply:
Right, I did this but forgot to add it to the proc. Let me add that quick